Library/Homebrew/github_packages.rb

```diff
@@ -315,28 +315,28 @@ def upload_bottle(user, token, skopeo, formula_full_name, bottle_hash, keep_old:
       tar_gz_sha256 = write_tar_gz(local_file, blobs)
 
       tab = tag_hash["tab"]
-      architecture = TAB_ARCH_TO_PLATFORM_ARCHITECTURE[tab["arch"].presence || bottle_tag.arch.to_s]
-      raise TypeError, "unknown tab['arch']: #{tab["arch"]}" if architecture.blank?
+      architecture = TAB_ARCH_TO_PLATFORM_ARCHITECTURE[tab.arch.presence || bottle_tag.arch.to_s]
+      raise TypeError, "unknown tab.arch: #{tab.arch}" if architecture.blank?
 
-      os = if tab["built_on"].present? && tab["built_on"]["os"].present?
-        BUILT_ON_OS_TO_PLATFORM_OS[tab["built_on"]["os"]]
+      os = if tab["built_on"].present? && tab.built_on["os"].present?
+        BUILT_ON_OS_TO_PLATFORM_OS[tab.built_on["os"]]
       elsif bottle_tag.linux?
         "linux"
       else
         "darwin"
       end
-      raise TypeError, "unknown tab['built_on']['os']: #{tab["built_on"]["os"]}" if os.blank?
+      raise TypeError, "unknown tab.built_on['os']: #{tab.built_on["os"]}" if os.blank?
 
-      os_version = tab["built_on"]["os_version"].presence if tab["built_on"].present?
+      os_version = tab.built_on["os_version"].presence if tab.built_on.present?
       case os
       when "darwin"
         os_version ||= "macOS #{bottle_tag.to_macos_version}"
       when "linux"
         os_version&.delete_suffix!(" LTS")
         os_version ||= OS::LINUX_CI_OS_VERSION
-        glibc_version = tab["built_on"]["glibc_version"].presence if tab["built_on"].present?
+        glibc_version = tab.built_on["glibc_version"].presence if tab.built_on.present?
         glibc_version ||= OS::LINUX_GLIBC_CI_VERSION
-        cpu_variant = tab["oldest_cpu_family"] || Hardware::CPU::INTEL_64BIT_OLDEST_CPU.to_s
+        cpu_variant = tab.built_on&.(:[], "oldest_cpu_family") || Hardware::CPU::INTEL_64BIT_OLDEST_CPU.to_s
```

Comment:
I think this was an earlier bug, it should be found in `built_on` with `glibc_version` above, as the two or grouped in https://github.com/Homebrew/brew/blob/master/Library/Homebrew/extend/os/linux/development_tools.rb#L51-L52